### PR TITLE
Cache repeated queries to ISS

### PR DESCRIPTION
**Pull request by DanielHeath**

### CODE REVIEW

File: src/iss.js [hunk 1]
```diff
@@ -193,10 +194,18 @@ async function attachTransportTimes(
     return services;
 }
 
+let requestObjectsCache = new Cache();
+
 export async function requestObjects(
     path: string,
     data: ?searchRequest,
 ): Promise<searchResults> {
+    let url_ = mungeUrlQuery(path, data);
```
Comment (danni) [inline src/iss.js]: const ?

File: src/iss.js [hunk 2]
```diff
@@ -193,10 +194,18 @@ async function attachTransportTimes(
...
+
+    if (requestObjectsCache.exactHit(url_)) {
+        return requestObjectsCache.exactHit(url_);
```
Comment (danni) [inline src/iss.js]: hit = blah, if hit return hit

File: src/iss/Cache.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import url from "url";
```
Comment (danni) [inline src/iss/Cache.js:3]: Use an existing library.

Comment (DanielHeath) [inline src/iss/Cache.js:3]: Done. Anything else?

**Comment by danni** [comment]
++

**Pull request merged**